src/Bicep.Core/Syntax/SyntaxRewriteVisitor.cs

```diff
@@ -0,0 +1,534 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Parser;
+
+namespace Bicep.Core.Syntax
+{
+    public abstract class SyntaxRewriteVisitor : ISyntaxVisitor
```

Comment:
>SyntaxRewriteVisitor [](start = 26, length = 20)

This is cool. I used the Roslyn equivalent once and it was great for modifying source files.